compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -360,6 +360,16 @@ impl<'tcx> Stable<'tcx> for FieldIdx {
     }
 }
 
+impl<'tcx> Stable<'tcx> for OffsetOfIdx {
+    type T = usize;
+    fn stable(&self, _: &mut Tables<'tcx>) -> Self::T {
+        match self {
+            OffsetOfIdx::Field(f) => f.as_usize(),
```

Comment:
I'm open to suggestions on what the SMIR should be for this. I think you can justify leaving it as `usize` only because in context it should always be clear whether an index is a variant index on an enum or a field index on a struct. But for future extensibility making `T` here an enum might be wise?